src/Lumper.UI/Views/MainWindow.axaml.cs

```diff
@@ -1,21 +1,74 @@
-using System;
-using System.ComponentModel;
-using Avalonia.Controls;
-using Lumper.UI.ViewModels;
-
-namespace Lumper.UI.Views;
-
-public partial class MainWindow : Window
-{
-    public MainWindow()
-    {
-        InitializeComponent();
-    }
-
-    private async void Window_OnClosing(object? sender, CancelEventArgs e)
-    {
-        if (DataContext is not MainWindowViewModel model)
-            throw new ArgumentOutOfRangeException();
-        await model.OnClose(e);
-    }
-}
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Input;
+using Lumper.UI.ViewModels;
+
+namespace Lumper.UI.Views;
+
+public partial class MainWindow : Window
+{
+    public MainWindow()
+    {
+        InitializeComponent();
+
+        AddHandler(DragDrop.DragOverEvent, DragOver);
+        AddHandler(DragDrop.DropEvent, Drop);
+    }
+
+    private async void Window_OnClosing(object? sender, CancelEventArgs e)
+    {
+        if (DataContext is not MainWindowViewModel model)
+            throw new ArgumentOutOfRangeException();
+        await model.OnClose(e);
+    }
+
+    private void DragOver(object? sender, DragEventArgs e)
+    {
+        e.DragEffects = DragDropEffects.Link;
+
+        var names = e.Data.GetFileNames() ?? new List<String>();
+
+        if (!e.Data.Contains(DataFormats.FileNames) || !names.FirstOrDefault("").ToLower().EndsWith(".bsp"))
+            e.DragEffects = DragDropEffects.None;
+    }
+
+    private void Drop(object? sender, DragEventArgs e)
+    {
+        if (e.Data.Contains(DataFormats.FileNames))
+        {
+            var names = e.Data.GetFileNames() ?? new List<String>();
+            foreach (string target in names)
+            {
+                if (!target.ToLower().EndsWith(".bsp"))
+                    continue;
+
+                if (((MainWindowViewModel)DataContext!).BspModel == null)

```

Comment:
There is too much logic and not enough UI here for it to be in `View`. Adding the drag and drop events in code-behind is fine if its easier but most of this can probably be moved to the `ViewModel`. At the very least the process spawning should be in the `ViewModel` in my opinion.
The ViewModel itself shouldn't know any avalonia specific things. (It does right now, but thats a me problem)